Source/WebDriver/WebDriverService.cpp

```diff
@@ -61,8 +61,18 @@ WebDriverService::WebDriverService()
     , m_browserTerminatedObserver([this](const String& sessionID) { onBrowserTerminated(sessionID); })
 #endif
 {
+#if ENABLE(WEBDRIVER_BIDI)
+    SessionHost::addBrowserTerminatedObserver(m_browserTerminatedObserver);
+#endif
 }
 
+#if ENABLE(WEBDRIVER_BIDI)
+WebDriverService::~WebDriverService()
+{
+    SessionHost::removeBrowserTerminatedObserver(m_browserTerminatedObserver);
+}
+#endif
+
```

Comment:
I think it's better to always define the destructor and move the ifdef inside. This way we don't need another ifdef in the header.